src/gpu/intel/ocl/kernel.cpp

```diff
@@ -219,12 +219,13 @@ status_t kernel_t::parallel_for(impl::stream_t &stream,
         OCL_CHECK(err);
         xpu::ocl::event_t::from(out_dep).events = {event};
     } else {
-        bool save_event = save_events_ || stream.is_profiling_enabled();
         cl_int err = clEnqueueNDRangeKernel(queue, *kernel, ndims, nullptr,
                 range.global_range().data(),
                 range.local_range() ? range.local_range().data() : nullptr, 0,
-                nullptr, save_event ? &event.unwrap() : nullptr);
+                nullptr, &event.unwrap());
```

Comment:
This makes `clEnqueueNDRangeKernel()` to always return an `event` which may have performance impact.

I think we don't really need an event without profiling enabled - as we can't query it anyway so the previous code, without changes, based on `save_event` should probably work here.